python/pyspark/mllib/linalg.py

```diff
@@ -821,6 +822,36 @@ def __reduce__(self):
             self.numRows, self.numCols, self.values.tostring(),
             int(self.isTransposed))
 
+    def __str__(self):
+        """
+        Pretty printing of a DenseMatrix
+
+        >>> dm = DenseMatrix(2, 2, range(4))
+        >>> print(dm)
+        0.0  2.0
+        1.0  3.0
+        >>> dm = DenseMatrix(2, 2, range(4), isTransposed=True)
+        >>> print(dm)
+        0.0  1.0
+        2.0  3.0
+        """
+        ds = []
+        for row in self.toArray():
+            ds.append("  ".join([str(col) for col in row]))
+        return "\n".join(ds)
+
+    def __repr__(self):
+        """
+        Representation of a DenseMatrix
+
+        >>> dm = DenseMatrix(2, 2, range(4))
+        >>> dm
+        DenseMatrix(2, 2, [0.0, 1.0, 2.0, 3.0], False)
+        """
+        entries = ', '.join([_format_float(val) for val in self.values])
```

Comment:
We shouldn't always display all elements in case there are millions of them. Again, it would be nice to reuse numpy's implementation.